oc-chef-pedant/spec/api/stats_spec.rb

```diff
@@ -0,0 +1,258 @@
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "erchef_pooler_queued_requestors",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "erchef_pooler_members_max",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "erchef_pooler_members_free",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "erchef_pooler_members_in_use",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "pg_stat_n_conns",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "pg_stat_n_active_conns",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "pg_stat_tidx_blks_hit",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_tidx_blks_read",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_toast_blks_hit",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_toast_blks_read",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_idx_blks_hit",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_idx_blks_read",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_heap_blocks_hit",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_heap_blocks_read",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_n_dead_tup",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "pg_stat_n_live_tup",
+        "type" => "GAUGE",
+      },
+      {
+        "name" => "pg_stat_n_tup_del",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_n_tup_upd",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_n_tup_ins",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_tup_fetch",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_idx_scan",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_seq_tup_read",
+        "type" => "COUNTER",
+      },
+      {
+        "name" => "pg_stat_seq_scan",
+        "type" => "COUNTER",
+      }
+    ]
+  end
+
+  let(:response_type_map) do
+    Hash[response_body.map { |stat| [stat["name"], stat["type"]] }]
+  end
+
+  let(:boolean_stats) { ["erlang_vm_time_correction", "erlang_vm_threads", "erlang_vm_smp_support"] }
+
+  # Figure out how to setup enable/disable basic-auth for test without chef-server-ctl reconfigure.
+  it "returns a list of collected statistics", :smoke do
+    get(request_url, platform.non_admin_user).should look_like({
+      :status => 200,
+      :body => response_body
+    })
+  end
+
+  it "returns metrics of the correct type", :smoke do
+    response = JSON.parse(get(request_url, platform.non_admin_user))
+    response_type_map.each do |name, type|
+      stat = response.find { |s| s["name"] == name }
+      expect(stat["metrics"]).not_to be_empty
+      stat["metrics"].each do |metric|
```

Comment:
instead of iterating inside the `it`, is it possible to create an `it` for each so that if there is a failure, you know what is broken?

Reply:
Something like https://coderwall.com/p/5kfxhg/dynamic-rspec-tests